lib/ansible/modules/cloud/docker/docker_swarm_info.py

```diff
@@ -305,6 +325,10 @@ def get_essential_facts_services(item):
 
         return object_essentials
 
+    def get_docker_swarm_unlock_key(self):
```

Comment:
This should be in `AnsibleDockerSwarmClient` class as well in my opinion and either unified for both `docker_swarm` and `docker_swarm_info` maybe?

Reply:
Please check https://github.com/ansible/ansible/pull/54490#discussion_r270585152
I think it makes more sense the way it is now: `docker_swarm` needs the structure with `UnlockKey` in it, while `docker_swarm_info` needs the raw value.f

Reply:
But it may be a universal method in the `AnsibleDockerSwarmClient` class where you add a parameter to define if you want raw value or structure. The process of reading key is shared between two modules that why it fits more into a shared library in my opinion. And in case a new module for locking/unlocking needs it as well it will add the third module to use it.

Reply:
I personally don't like functions with variable return type (depending on an argument). It's just impossible to verify code using such functions with static code analysis.

I would not start adding methods to module_utils/docker/swarm.py just because we might need them in the future, but do that refactoring when we actually add a new module that needs them. Then we'll find out what exactly we need.

Reply:
In my proposal, it is a method shared already between two modules. But of course, we can move and consolidate later

Reply:
My proposal is to wait until we implement the locking-module and see what kind of implementation is to be used there (dict with `UnlockKey` or just a string). The common implementation can then be lifted to `AnsibleDockerSwarmClient` and the other one can be used as an override.

Reply:
Ok, sounds good